### ISSUE TITLE
Would you consider to add some CV examples with hivemind?[Feature Request]

### ISSUE THREAD

**Issue by elricwan** [learning-at-home/hivemind#500]
**Is your feature request related to a problem? Please describe.**

The CV models usually have the batch normalization part, which might cause problems when using hivemind, would you consider apply the framework to the CV model?

**Comment by borzunov** [comment]
Hi @elricwan!

We're really interested in a CV example. Actually, we had a contributor who worked on a CV example (a vision transformer) for the repository, but unfortunately he didn't finish the project.

We believe there are no conceptual roadblocks to training CV models with hivemind. In particular, batch norm is not a problem because:

1. Many CV models work well without batch norm:
    - **Transformer-based** CV models (including vision transformers) don't use batch norm at all, they use instance norm instead.
    - Batch norm in ResNet can be replaced with **fixup init** (or newer methods similar to it).

2. If you'd still like to have batch norm:
    - While you can't perform batch norm with a global batch (= the batch spread over all peers) out-of-the-box in hivemind, **you can perform it with a local micro-batch** (= the batch of samples processed on a peer's GPU in parallel). Usually, convolutional NNs are small enough to fit a micro-batch of >= 32 samples into a peer's GPU memory - this should give good enough statistics.
    - If you'd still like to perform "honest" batch norm on a global batch, you can average the batch norm stats in the `hivemind.Optimizer`s **extra_tensors**.

**Comment by elricwan** [comment]
Hi borzunov,

Thank you for the detailed response, I would try without batch normalization first. And I am also looking forward to the CV example in the future.
